Source/Streamstone.Tests/Scenarios/Tracking_entity_changes.cs

```diff
@@ -577,15 +586,21 @@ yield return
             }
         }
 
-        void InsertTestEntity(TestEntity entity)
+        void InsertTestEntity(ITableEntity entity)
```

Comment:
Those tests were testing in-memory change tracking behavior. If I change it to:

```csharp
[Test]
public void When_Insert_Or_Merge_followed_by_Insert_Or_Merge()
{
    var entity = new ExtendedTestEntity(EntityRowKey)
    {
        Data = "zzz"
    };

    var extendedEntity = new ExtendedTestEntity(EntityRowKey)
    {
        AdditionalData = "zzz",
    };

    var events = new[]
    {
        CreateEvent(Include.InsertOrMerge(entity)),
        CreateEvent(Include.InsertOrMerge(extendedEntity))
    };

    Stream.Write(stream, events);

    var stored = RetrieveEntity<ExtendedTestEntity>(EntityRowKey);
    Assert.That(stored.Data, Is.EqualTo("zzz"));
    Assert.That(stored.AdditionalData, Is.EqualTo("zzz"));
}
```
it blows with InvalidOperationException "Mistaken identity. Different instance of entity has been already registered in this session under RowKey: INV-0001" :(

`ChangeTracker` maintains by-ref identity of entities. This was done to have Etag flow between operations. I think this could be changed to by-value tracking and then Etag would need to be propagated (copied) automatically by the tracker.

I can't remember but there were some other reasons I did by-ref entity identity tracking. If I won't recall until morning we can change that logic to make this test pass.